oxidation/crates/parsec_api_crypto/src/rustcrypto/private.rs

```diff
@@ -130,65 +126,49 @@ impl PrivateKey {
     }
 
     pub fn generate_shared_secret_key(&self, peer_public_key: &PublicKey) -> SecretKey {
-        SecretKey::from(x25519(
-            *self.0.as_bytes(),
-            peer_public_key.0.as_bytes().to_owned(),
-        ))
+        SecretKey::from(x25519(*self.0.as_bytes(), *peer_public_key.0.as_bytes()))
     }
 }
 
 impl AsRef<[u8]> for PrivateKey {
-    #[inline]
     fn as_ref(&self) -> &[u8] {
-        // Unsafe on private key, what could possibly go wrong ?
-        // TODO: crypto_box::SecretKey should implement AsRef<[u8]> instead
-        unsafe {
-            ::std::slice::from_raw_parts(
-                (self as *const Self) as *const u8,
-                ::std::mem::size_of::<Self>(),
-            )
-        }
+        self.0.as_bytes()
```

Comment:
So `crypto_box::SecretKey` don't implement `AsRef<[u8]>`, hence the crazy unsafe block, in fact but we missed the `crypto_box::SecretKey.as_bytes()` that was what we needed ?

Reply:
Yes it is the same thing